src/cascadia/TerminalSettingsEditor/Appearances.cpp

```diff
@@ -257,6 +257,20 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
                     _PropertyChangedHandlers(*this, PropertyChangedEventArgs{ L"UsingMonospaceFont" });
                 }
             });
+
+            // make sure to send all the property changed events once here
+            // we do this in the case an old appearance was deleted and then a new one is created,
+            // the old settings need to be updated in xaml
```

Comment:
Hmm... could we somehow detect something like "AppearanceChanged"?

Reply:
That's what's happening here already! (Note that we only get here from the `_ViewModelChanged` event)

The thing is the `Appearances` object (not the view model) has some settings that xaml binds to (like `CurrentColorScheme`) that hang around even if the view model is deleted, then when a new view model is created we need to tell xaml that these settings may have changed